crates/voicevox_core/src/c_export.rs

```diff
@@ -254,13 +257,113 @@ pub extern "C" fn voicevox_load_openjtalk_dict(dict_path: *const c_char) -> Voic
     result_code
 }
 
+#[no_mangle]
+pub extern "C" fn voicevox_audio_query(
+    text: *const c_char,
+    speaker_id: i64,
+    output: *mut *mut c_char,
+) -> VoicevoxResultCode {
+    let text = unsafe { CStr::from_ptr(text) };
+
+    let audio_query = &match audio_query(text, speaker_id, Internal::voicevox_audio_query) {
+        Ok(audio_query) => audio_query,
+        Err(result_code) => return result_code,
+    };
+
+    unsafe {
+        write_json_to_ptr(output, audio_query);
+    }
+    VoicevoxResultCode::VOICEVOX_RESULT_SUCCEED
+}
+
+#[no_mangle]
+pub extern "C" fn voicevox_audio_query_from_kana(
+    text: *const c_char,
+    speaker_id: i64,
+    output: *mut *mut c_char,
+) -> VoicevoxResultCode {
+    let text = unsafe { CStr::from_ptr(text) };
+
+    let audio_query = &match audio_query(text, speaker_id, Internal::voicevox_audio_query_from_kana)
+    {
+        Ok(audio_query) => audio_query,
+        Err(result_code) => return result_code,
+    };
+
+    unsafe {
+        write_json_to_ptr(output, audio_query);
+    }
+    VoicevoxResultCode::VOICEVOX_RESULT_SUCCEED
+}
+
+fn audio_query(
+    japanese_or_kana: &CStr,
+    speaker_id: i64,
+    method: fn(&mut Internal, &str, usize) -> Result<AudioQueryModel>,
+) -> std::result::Result<CString, VoicevoxResultCode> {
+    let japanese_or_kana = ensure_utf8(japanese_or_kana)?;
+    let speaker_id = speaker_id as usize;
+
+    let (audio_query, result_code) =
+        convert_result(method(&mut lock_internal(), japanese_or_kana, speaker_id));
+    let audio_query = audio_query.ok_or(result_code)?;
+    Ok(CString::new(audio_query.to_json()).expect("should not contain '\\0'"))
+}
+
+unsafe fn write_json_to_ptr(output_ptr: *mut *mut c_char, json: &CStr) {
+    let n = json.to_bytes_with_nul().len();
+    let json_heap = libc::malloc(n);
+    libc::memcpy(json_heap, json.as_ptr() as *const c_void, n);
+    output_ptr.write(json_heap as *mut c_char);
+}
+
 unsafe fn write_wav_to_ptr(output_wav_ptr: *mut *mut u8, output_size_ptr: *mut c_int, data: &[u8]) {
     output_size_ptr.write(data.len() as c_int);
     let wav_heap = libc::malloc(data.len());
     libc::memcpy(wav_heap, data.as_ptr() as *const c_void, data.len());
     output_wav_ptr.write(wav_heap as *mut u8);
 }
 
+#[no_mangle]
+pub extern "C" fn voicevox_synthesis(
+    audio_query: *const c_char,
```

Comment:
```suggestion
    audio_query_json: *const c_char,
```
jsonである旨は引数名で表現したほうが良いかと